lib/internal/http2/compat.js

```diff
@@ -330,45 +306,94 @@ class Http2ServerResponse extends Stream {
     code |= 0;
     if (code >= 100 && code < 200)
       throw new RangeError('Informational status codes cannot be used');
-    if (code < 200 || code > 999)
+    if (code < 200 || code >= 600)
       throw new RangeError(`Invalid status code: ${code}`);
     state.statusCode = code;
   }
 
+  addTrailers(headers) {
+    for (var name of Object.keys(headers)) {
```

Comment:
Does V8 5.8, [as tested with Chrome v58](https://jsperf.co/for-vs-foreach-vs-for-of), not eliminate the performance difference?

<img width="320" alt="chrome for/for-of benchmark" src="https://cloud.githubusercontent.com/assets/78718/25647584/bf011d3c-2ff5-11e7-974c-d44a72e9fef3.png">

I can still see a 20% advantage in microbenchmarks on Node.js (this fork) with V8 5.7 using `for` loop over `for-of`. Gist: https://gist.github.com/sebdeckers/f110f4c5f65fc70da39587cb5d0fba68

```
for x 29,955,989 ops/sec ±1.58% (81 runs sampled)
for-of x 22,832,152 ops/sec ±1.64% (78 runs sampled)
```